nmdc_runtime/site/translation/neon_translator.py

```diff
@@ -0,0 +1,757 @@
+            FROM soil_biosamples_combined
+            LEFT JOIN ntr_internalLab ON soil_biosamples_combined.sampleID = ntr_internalLab.sampleID
+            LEFT JOIN ntr_externalLab ON soil_biosamples_combined.sampleID = ntr_externalLab.sampleID
+        """
+        soil_biosamples_combined_ntr = pd.read_sql_query(query, self.conn)
+        soil_biosamples_combined_ntr.to_sql(
+            "soil_biosamples_combined_ntr", self.conn, if_exists="replace", index=False
+        )
+
+        query = """
+            SELECT sbcbn.*, net.neon_nlcd_value, net.envo_id, net.envo_label, net.env_local_scale
+            FROM soil_biosamples_combined_ntr sbcbn
+            LEFT JOIN neonEnvoTerms net ON sbcbn.nlcdClass = net.neon_nlcd_value
+        """
+        soil_biosamples_envo = pd.read_sql_query(query, self.conn)
+        soil_biosamples_envo.to_sql(
+            "soil_biosamples_envo", self.conn, if_exists="replace", index=False
+        )
+
+        query = """
+            SELECT dnaSampleID, startDate, collectDate, GROUP_CONCAT(sampleID, '|') AS sampleIDs
+            FROM soil_biosamples_envo
+            GROUP BY dnaSampleID
+        """
+        mg_pooling_table = pd.read_sql_query(query, self.conn)
+        pooling_ids_dict = (
+            mg_pooling_table.set_index("dnaSampleID")["sampleIDs"]
+            .str.split("|")
+            .to_dict()
+        )
+
+        query = """
+            SELECT dnaSampleID, GROUP_CONCAT(rawDataFilePath, '|') AS rawDataFilePaths
+            FROM neonRawDataFile
+            GROUP BY dnaSampleID
+        """
+        neon_raw_data_files = pd.read_sql_query(query, self.conn)
+        neon_raw_data_files_dict = (
+            neon_raw_data_files.set_index("dnaSampleID")["rawDataFilePaths"]
+            .str.split("|")
+            .to_dict()
+        )
+
+        query = """
+            SELECT dnaSampleID, genomicsSampleID, collectDate, laboratoryName, processedDate, sampleMass, qaqcStatus
+            FROM soil_biosamples_envo
+            GROUP BY genomicsSampleID
+        """
+        extraction_table = pd.read_sql_query(query, self.conn)
+
+        query = """
+            SELECT 
+                mms_metagenomeDnaExtraction.dnaSampleID, 
+                mms_metagenomeDnaExtraction.genomicsSampleID, 
+                mms_metagenomeDnaExtraction.sequenceAnalysisType,
+                mms_metagenomeDnaExtraction.laboratoryName,
+                mms_metagenomeDnaExtraction.collectDate,
+                mms_metagenomeDnaExtraction.processedDate,
+                mms_metagenomeSequencing.sequencingFacilityID,
+                mms_metagenomeSequencing.ncbiProjectID
+            FROM mms_metagenomeSequencing 
+            LEFT JOIN mms_metagenomeDnaExtraction ON mms_metagenomeDnaExtraction.dnaSampleID = mms_metagenomeSequencing.dnaSampleID
+        """
+        library_preparation_table = pd.read_sql_query(query, self.conn)
+        omics_processing_table = pd.read_sql_query(query, self.conn)
```

Comment:
Duplicate query?

Reply:
It's the same starting query from which we'll be creating the `LibraryPreparation` records and the `OmicsProcessing` records.